var/spack/repos/builtin/packages/py-pillow/package.py

```diff
@@ -94,8 +86,8 @@ def patch(self):
 
         if '+jpeg' in spec:
             setup.filter('JPEG_ROOT = None',
-                         'JPEG_ROOT = ("{0}", "{1}")'.format(
-                             spec['jpeg'].prefix.lib,
+                         'JPEG_ROOT=("{0}","{1}")'.format(
+                             spec['jpeg'].prefix.lib64,
```

Comment:
You should use:
```python
spec['jpeg'].libs.directories[0]
```
This will search the installation prefix for the correct libraries, regardless of whether they live in `lib` or `lib64`, and regardless of how they are named.

Reply:
What's the difference between ```.prefix.lib``` and ```.libs.directories``` ? 
Did spack change the way it looks for libraries since the original recipe for py-pillow was written ? Just curious as to why the original file used one instead of the other if the second one was more general.

Reply:
> What's the difference between `.prefix.lib` and `.libs.directories`?

`prefix` and `prefix.lib` are objects of type `Prefix`. They are basically strings with the magical property that `foo.bar.baz` will result in the string `/foo/bar/baz`, nothing special going on here. See [here](http://spack.readthedocs.io/en/latest/spack.util.html#module-spack.util.prefix) for docs.

`libs` is an object of type `LibraryList`. It is a powerful tool for managing libraries. You can ask for all kinds of things, from `libs.ld_flags` to `libs.names`. The default implementation recursively searches the installation prefix for a library with the same name as the package. Packages can override this if their libraries are named differently. See [here](http://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.LibraryList) for docs.

> Did spack change the way it looks for libraries since the original recipe for py-pillow was written?

Yep, this `libs` stuff was a little over a year ago maybe?